libraries/objectDefaultFiles/object.js

```diff
@@ -1959,6 +1960,22 @@
 
         this.changeToolSize = this.changeFrameSize;
 
+        let windowResizedCallbackCount = 0;
+        this.onWindowResized = function(callback) {
+            windowResizedCallbackCount++;
+            spatialObject.messageCallBacks[`onWindowResizedCall${windowResizedCallbackCount}`] = (msgContent) => {
```

Comment:
I agree, something along those lines is probably better. I was investigating why my current method works, and it's because in the current messageCallbacks implementation, whenever *any* message is posted into the iframe, *all* message callbacks will be triggered, from the code on object.js line 212:

```
    window.addEventListener('message', function (MSG) {
        if (!MSG.data) { return; }
        if (typeof MSG.data !== 'string') { return; }
        var msgContent = JSON.parse(MSG.data);
        for (var key in spatialObject.messageCallBacks) {
            spatialObject.messageCallBacks[key](msgContent);
        }
    }, false);
```

It is then the responsibility of each function to check whether `if (typeof msgContent[THIS_CALLBACK_NAME] !== 'undefined')`. So in my case, it doesn't matter what ID the messageCallbacks[ID] uses, as long as the callback has the correct msgContent check inside of it. We might want to consider optimizing this whole system, to minimize how many callbacks get (partially) triggered in response to each post message into the iframe.